plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java

```diff
@@ -65,13 +65,11 @@ public List<Route> routes() {
     public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         String algorithm = request.param(PARAMETER_ALGORITHM);
         String modelId = getParameterId(request, PARAMETER_MODEL_ID);
-        if (modelId == null) {
-            throw new IllegalArgumentException("model id is null");
-        }
 
         if (algorithm != null) {
+            MLPredictionTaskRequest mlPredictionTaskRequest = getRequest(modelId, algorithm, request);
             return channel -> client
-                .execute(MLPredictionTaskAction.INSTANCE, getRequest(modelId, algorithm, request), new RestToXContentListener<>(channel));
+                .execute(MLPredictionTaskAction.INSTANCE, mlPredictionTaskRequest, new RestToXContentListener<>(channel));
```

Comment:
Is this because the JVM complier is not able to dynamic resolve getRequest() before executing  client.execute()? If that's the case, we need to use another local variable for line 83. https://github.com/opensearch-project/ml-commons/blob/2.x/plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java#L83

Reply:
just for easy testing